lib/ensure/symlink-type.js

```diff
@@ -1,22 +1,25 @@
 'use strict'
 
-const fs = require('graceful-fs')
+const fs = require('../fs')
+const u = require('universalify').fromPromise
 
-function symlinkType (srcpath, type, callback) {
-  callback = (typeof type === 'function') ? type : callback
-  type = (typeof type === 'function') ? false : type
-  if (type) return callback(null, type)
-  fs.lstat(srcpath, (err, stats) => {
-    if (err) return callback(null, 'file')
-    type = (stats && stats.isDirectory()) ? 'dir' : 'file'
-    callback(null, type)
-  })
-}
+async function symlinkType (srcpath, type = false) {
```

Comment:
Not sure why the old logic worked this way, but no reason to default to `false` here.
```suggestion
async function symlinkType (srcpath, type) {
```

Reply:
Fixed in [`fbfc9c7` (#1023)](https://github.com/jprichardson/node-fs-extra/pull/1023/commits/fbfc9c7d86925105cf905df9e11b51132be386d4).